pkg/vsphere/disk/lazy_detach_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package disk
+
+import (
+	"context"
+	"path"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/datastore"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+)
+
+// TestLazyDetach tests lazy detach functionality to make sure that every ESXi version shows this behaviour
+// https://github.com/vmware/vic/issues/5565
+func TestLazyDetach(t *testing.T) {
+	log.SetLevel(log.DebugLevel)
+
+	client := Session(context.Background(), t)
+	if client == nil {
+		return
+	}
+
+	imagestore := &object.DatastorePath{
+		Datastore: client.Datastore.Name(),
+		Path:      datastore.TestName("lazyReconfigure"),
+	}
+
+	fm := object.NewFileManager(client.Vim25())
+
+	// create a directory in the datastore
+	// eat the error because we dont care if it exists
+	fm.MakeDirectory(context.TODO(), imagestore.String(), nil, true)
+	// Nuke the image store
+	defer func() {
+		task, err := fm.DeleteDatastoreFile(context.TODO(), imagestore.String(), nil)
```

Comment:
imagestore path is constructed via  datastore.TestName("lazyReconfigure") which creates a random string and concatenates the result with the passed parameter